src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -5785,13 +5784,17 @@ int SketchObject::exposeInternalGeometry(int GeoId)
         this->addGeometry(igeo, true);
         this->addConstraints(icon);
 
-        for (std::vector<Part::Geometry*>::iterator it = igeo.begin(); it != igeo.end(); ++it)
-            if (*it)
-                delete *it;
+        for (auto* geo : igeo) {
+            if (geo) {
+                delete geo;
+            }

```

Comment:
`clang-tidy` suggests this check can be removed. Deleting a `nullptr` does nothing. That said it's not a major concern.

Reply:
Why do you say that 'Deleting a nullptr does nothing' ? This is deleting geo if it is not nullptr.

Besides this PR is just simplifying the syntax. If this needs to be removed it should be a separate PR.

Reply:
I mean this can be replaced with:
```cpp
for (auto geo : ...) {
    delete geo;
}
```